internal/services/mysql/mysql_configuration_resource.go

```diff
@@ -24,6 +24,8 @@ func resourceMySQLConfiguration() *pluginsdk.Resource {
 		Read:   resourceMySQLConfigurationRead,
 		Delete: resourceMySQLConfigurationDelete,
 
+		DeprecationMessage: "The `azurerm_mysql_configuration` resource will be deprecated since Azure Database for MySQL - Single Server and its sub resources are scheduled for retirement (https://go.microsoft.com/fwlink/?linkid=2216041) by September 16, 2024 and they migrate (https://go.microsoft.com/fwlink/?linkid=2202255) to Azure Database for MySQL - Flexible Server and its sub resources now. Please use the `azurerm_mysql_flexible_server_configuration` resource instead.",
```

Comment:
Can you please apply the suggestion made above to all of the deprecation messages in this PR

Reply:
Updated to all.